packages/cactus-plugin-ledger-connector-iroha/src/test/typescript/integration/run-transaction-endpoint-v1.test.ts

```diff
@@ -0,0 +1,783 @@
+
+import { Configuration } from "@hyperledger/cactus-core-api";
+
+const testCase = "runs tx on an Iroha v1.2.0 ledger";
+const logLevel: LogLevelDesc = "TRACE";
+
+test.onFailure(async () => {
+  await Containers.logDiagnostics({ logLevel });
+});
+
+test("BEFORE " + testCase, async (t: Test) => {
+  const pruning = pruneDockerAllIfGithubAction({ logLevel });
+  await t.doesNotReject(pruning, "Pruning didn't throw OK");
+  t.end();
+});
+
+test(testCase, async (t: Test) => {
+  const postgres = new PostgresTestContainer({
+    containerImageName: "postgres",
+    containerImageVersion: "9.5-alpine",
+    postgresPort: 5432,
+    envVars: ["POSTGRES_USER=postgres", "POSTGRES_PASSWORD=mysecretpassword"],
+  });
+
+  const iroha = new IrohaTestLedger({
+    containerImageVersion: "1.20c",
+    containerImageName: "hanxyz/iroha",
+    rpcToriiPort: 50051,
+    //logLevel: "TRACE",
+    envVars: [
+      "IROHA_POSTGRES_HOST=postgres_1",
+      "IROHA_POSTGRES_PORT=5432",
+      "IROHA_POSTGRES_USER=postgres",
+      "IROHA_POSTGRES_PASSWORD=mysecretpassword",
+      "KEY=node0",
+    ],
+  });
+
+  await postgres.start(); //start postgres first
+  await iroha.start();
+  test.onFinish(async () => {
+    await iroha.stop();
+    await iroha.destroy();
+    await postgres.stop();
+    await postgres.destroy();
+  });
+
+  const rpcToriiPortHost = await iroha.getRpcToriiPortHost();
+  const factory = new PluginFactoryLedgerConnector({
+    pluginImportType: PluginImportType.Local,
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const connector: PluginLedgerConnectorIroha = await factory.create({
+    rpcToriiPortHost,
+    instanceId: uuidv4(),
+    pluginRegistry: new PluginRegistry(),
+  });
+
+  const expressApp = express();
+  expressApp.use(bodyParser.json({ limit: "250mb" }));
+  const server = http.createServer(expressApp);
+  const listenOptions: IListenOptions = {
+    hostname: "0.0.0.0",
+    port: 0,
+    server,
+  };
+  const addressInfo = (await Servers.listen(listenOptions)) as AddressInfo;
+  test.onFinish(async () => await Servers.shutdown(server));
+  const { address, port } = addressInfo;
+  const apiHost = `http://${address}:${port}`;
+  const apiConfig = new Configuration({ basePath: apiHost });
+  const apiClient = new IrohaApi(apiConfig);
+
+  await connector.getOrCreateWebServices();
+  await connector.registerWebServices(expressApp);
+
+  let firstTxHash;
+  {
+    const req = {
+      commandName: "createAccount",
+      params: [
+        "user1",
+        "test",
+        "fe31d7bc2dbe695b1cd5c706dde902d5fb6e6ce4b60e08418c842fdef5805230", //node4.pub
+      ],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
+  }
+
+  {
+    const req = {
+      commandName: "getAccount",
+      params: ["admin@test"],
```

Comment:
@hxlaf If these credentials are somehow hardcoded in the container image then it would be good to have a dedicated method on the `IrohaTestLedger` class to retrieve them, one for the username and one for the password. That way if we ever change the credentials in the image we won't have to go through all the tests hunting for these [magic strings](https://en.wikipedia.org/wiki/Magic_string).

Reply:
Sure! I am working on that.